go/api/base_client.go

```diff
@@ -948,3 +948,27 @@ func (client *baseClient) Del(keys []string) (Result[int64], error) {
 
 	return handleLongResponse(result)
 }
+
+func (client *baseClient) Dump(key string) (Result[string], error) {
+	result, err := client.executeCommand(C.Dump, []string{key})
+	if err != nil {
+		return CreateNilStringResult(), err
+	}
+	return handleStringOrNullResponse(result)
+}
+
+func (client *baseClient) ObjectEncoding(key string) (Result[string], error) {
+	result, err := client.executeCommand(C.ObjectEncoding, []string{key})
+	if err != nil {
+		return CreateNilStringResult(), err
+	}
+	return handleStringOrNullResponse(result)
+}
+
+func (client *baseClient) Restore(key string, ttl int64, value string) (Result[string], error) {
```

Comment:
According to Valkey command guide (https://valkey.io/commands/restore/), `restore` command has optional arguments of `replace`, `absttl`, `IDLETIME`, and `FREQ`. Please add different function signatures in this PR.

Reply:
Restore Options is now added to thi commit cfcb0e893a0099380b4e50b42ee02721fdf970c8